lib/routes/sina/tousu.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * @Descripttion: description
+ * @Author: -0V
+ * @Date: 2024-11-01 10:59:51
+ * @LastEditTime: 2024-11-07 16:11:14
+ */
+import { Route } from '@/types';
+
+import logger from '@/utils/logger';
+import puppeteer from '@/utils/puppeteer';
+import { load } from 'cheerio';
+
+export const route: Route = {
+    path: '/tousu',
+    name: '黑猫投诉',
+    url: 'tousu.sina.com.cn',
+    example: '/sina/tousu',
+    maintainers: ['JianLinWei1'],
+    radar: [
+        {
+            source: ['tousu.sina.com.cn'],
+        },
+    ],
+    handler,
+};
+
+async function handler() {
+    const browser = await puppeteer();
+    const page = await browser.newPage();
+    const link = 'https://tousu.sina.cn';
+    logger.http(`Requesting ${link}`);
+    await page.goto(link, {
+        waitUntil: 'networkidle2',
+    });
+    const response = await page.content();
+    page.close();
+    const $ = load(response);
```

Comment:
Unnecessary use of puppeteer. You should [prioritise using APIs](https://docs.rsshub.app/joinus/new-rss/start-code#via-api) to obtain data instead of using puppeteer.

The API endpoint is `https://tousu.sina.cn/api/index/feed` and you can use query like
```ts
const generateRandomString = (length) => {
    let result = '';
    const characters = '0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ';
    for (let index = 0; index < length; index++) {
        result += characters[Math.round(Math.random() * (characters.length - 1))];
    }
    return result;
};

const query = {
    ts: Date.now(),
    rs: generateRandomString(16),
    type: 1,
    page_size: 10,
    page: 1,
    };

const response = await ofetch(`${link}/api/index/feed`, {
    query: {
        ...query,
        signature: crypto
            .createHash('sha256')
            .update(['$d6eb7ff91ee257475%', ...Object.values(query)].sort().join(''))
            .digest('hex'),
        },
    });
```

and the data is available in `response.result.data.lists`